lms/djangoapps/instructor/views/api.py

```diff
@@ -1388,6 +1397,47 @@ def validator(file_storage, file_to_validate):
     return JsonResponse()
 
 
+@method_decorator(transaction.non_atomic_requests, name='dispatch')
```

Comment:
I don't think ```transaction.non_atomic_requests``` is necessary here or on the ``add_users_to_cohorts`` function, because ```cohorts.add_user_to_cohort``` is atomic. 

If the tests pass without this decorator, I think it's best to remove it, to avoid future confusion.

Reply:
@davestgermain, this is required because the CSV import is done using a celery task and the task submission code uses `outer_atomic()`.

Reply:
I'm not sure if it's needed on the celery task, either. But rather than trying to figure that out, could you just put a brief comment here explaining that the non_atomic_requests decorator is required because it's calling the particular celery task?